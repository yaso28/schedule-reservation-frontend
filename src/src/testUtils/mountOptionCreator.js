import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
