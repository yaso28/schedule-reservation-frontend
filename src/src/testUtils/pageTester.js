import { mount } from '@vue/test-utils'
import MyPage from '@/components/containers/MyPage'

class PageTester {
  async mount (pageComponent, route, mountOption) {
    await mountOption.router.push(route)
    const pageWrapper = mount(MyPage, mountOption)
    return pageWrapper.findComponent(pageComponent)
  }

  assertTitleTag () {
    expect(document.title).toMatchSnapshot()
  }

  assertMetaTags () {
    expect(document.querySelectorAll('meta[data-vue-meta]')).toMatchSnapshot()
  }
}

export default new PageTester()
