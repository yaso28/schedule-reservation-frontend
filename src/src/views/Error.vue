<template>
  <div>
    <CAlert color="danger">
      {{ $t(`message.error.${code}`) }}
    </CAlert>
  </div>
</template>

<script>
export default {
  name: 'MyError',
  props: {
    code: {
      type: String,
      required: true
    }
  }
}
</script>
