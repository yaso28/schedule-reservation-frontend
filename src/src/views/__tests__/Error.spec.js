import Error from '../Error'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import pageTester from '@/testUtils/pageTester'
import storeModules from '@/store/modules'

jest.mock('@/store/modules')

describe('Error', () => {
  let mountOption
  beforeEach(() => {
    mountOption = mountOptionCreator.create(storeModules)
  })

  describe.each(['403', '404'])('%s', (code) => {
    describe('マウント', () => {
      let wrapper
      beforeEach(async () => {
        wrapper = await pageTester.mount(Error, {
          name: 'error',
          params: { code }
        }, mountOption)
      })

      test('title', () => {
        pageTester.assertTitleTag()
      })

      test('meta tags', () => {
        pageTester.assertMetaTags()
      })

      test('ページ全体表示', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
