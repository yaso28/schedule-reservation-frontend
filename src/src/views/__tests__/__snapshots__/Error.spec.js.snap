// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error 403 マウント meta tags 1`] = `
NodeList [
  <meta
    content="noindex"
    data-vmid="robots"
    data-vue-meta="2"
    name="robots"
  />,
]
`;

exports[`Error 403 マウント title 1`] = `"やそスポーツクラブ"`;

exports[`Error 403 マウント ページ全体表示 1`] = `
<div>
  <transition-stub
    appear="true"
    data-v-c95d1f08=""
    name="fade"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      class="alert alert-danger"
      data-v-c95d1f08=""
      role="alert"
    >
      <!---->
      
    権限がありません。
  
    </div>
  </transition-stub>
</div>
`;

exports[`Error 404 マウント meta tags 1`] = `
NodeList [
  <meta
    content="noindex"
    data-vmid="robots"
    data-vue-meta="5"
    name="robots"
  />,
]
`;

exports[`Error 404 マウント title 1`] = `"やそスポーツクラブ"`;

exports[`Error 404 マウント ページ全体表示 1`] = `
<div>
  <transition-stub
    appear="true"
    data-v-c95d1f08=""
    name="fade"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      class="alert alert-danger"
      data-v-c95d1f08=""
      role="alert"
    >
      <!---->
      
    URLに誤りがあります。
  
    </div>
  </transition-stub>
</div>
`;
